WordPress/Classes/System/WordPressAppDelegate.swift

```diff
@@ -885,17 +885,21 @@ extension WordPressAppDelegate {
         }
 
         // Get the Apple User ID state. If not authorized, log out the account.
-        WordPressAuthenticator.shared.checkAppleIDCredentialState(for: appleUserID) { [weak self] (authorized, error) in
+        WordPressAuthenticator.shared.getAppleIDCredentialState(for: appleUserID) { [weak self] (state, error) in
 
-            // An error exists only for the 'not found' state.
-            // 'not found' is a valid state when logging in with an Apple account for the first time.
-            if let error = error {
-                DDLogDebug("checkAppleIDCredentialState: Apple ID state not found: \(error.localizedDescription)")
-            }
+            DDLogDebug("checkAppleIDCredentialState: Apple ID state: \(state.rawValue)")
 
-            if !authorized {
-                DDLogInfo("checkAppleIDCredentialState: Unauthorized Apple ID. User signed out.")
+            switch state {
+            case .revoked:
+                DDLogInfo("checkAppleIDCredentialState: Revoked Apple ID. User signed out.")
                 self?.logOutDefaultWordPressComAccount()
```

Comment:
We should also ensure we delete the Apple ID from the keychain on log out. I noticed an issue with my device where it somehow still has the Apple ID in the keychain, even though I logged out and logged back in with a different account. Because of this, each time we do the apple credential state check here, the app sees I have a revoked ID and logs me out again.

Reply:
Hey @frosty . Nice catch! It has been done. See what you think now. Thanks!